src/Extensions.cs

```diff
@@ -232,5 +240,57 @@ internal static void Add(this LocalVariablesEncoder sig, LocalVariableInfo local
                 typeEncoder.Type(typeHandler, type.IsValueType);
             }
         }
+
+#if NETSTANDARD || NET46
+        internal static bool IsGenericMethodParameter(this Type type) => type.IsGenericParameter && type.DeclaringMethod != null;
```

Comment:
In order to reduce the number of `#if` sections in this pull request to just one, I would be willing to replace all occurrences of `Type.IsGenericMethodParameter` and `MethodBase.IsConstructedGenericMethod` with calls to these two extension methods, and provide a trivial implementation for these two extension methods on .NET Core.

Reply:
Done.